application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/websocket/WebsocketServer.java

```diff
@@ -0,0 +1,152 @@
+package org.openjdk.jmc.flightrecorder.ui.websocket;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketAdapter;
+import org.eclipse.jetty.websocket.server.NativeWebSocketServletContainerInitializer;
+import org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter;
+import org.openjdk.jmc.common.item.IItemCollection;
+import org.openjdk.jmc.flightrecorder.serializers.json.IItemCollectionJsonSerializer;
+
+public class WebsocketServer {
+	private static int PORT = 8029;
+	private static int MAX_MESSAGE_SIZE = 1024 * 1024 * 1024;
+
+	private List<WebSocketConnectionHandler> handlers = new ArrayList<>();
+	private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+	private IItemCollection currentSelection = null;
+
+	public WebsocketServer() {
+		executorService.execute(() -> startServer());
+	}
+
+	private void startServer() {
+		Server server = new Server();
+		ServerConnector connector = new ServerConnector(server);
+		connector.setPort(PORT);
```

Comment:
I'm not quite sure how this feature will be used, but can it make sense to only bind to locahost ?

Reply:
:+1: Yup, definitely! This was in the plan :wink: it's done now.